### PR TITLE
GitHub Actions - workflow `Connector Integration Tests` - add retry

### LINKED ISSUE #14450: GitHub Actions - workflow `Connector Integration Tests` - jobs can fail AWS RequestLimitExceeded
## Tell us about the problem you're trying to solve
Workflow: `Connector Integration Tests` - runs another workflow `integration-test` for all source and destination connectors. Because a lot of workflows are run for short period of time some workflow_runs failed with rate-limit from AWS.
For example for `destination-s3`:

```
2022-07-05T11:08:55Z https://github.com/airbytehq/airbyte/actions/runs/2615718248 (OK)
2022-07-04T11:08:18Z https://github.com/airbytehq/airbyte/actions/runs/2609646320 (RequestLimitExceeded: Request limit exceeded)
2022-07-03T11:09:24Z https://github.com/airbytehq/airbyte/actions/runs/2604848973 (RequestLimitExceeded: Request limit exceeded)
2022-07-02T11:08:21Z https://github.com/airbytehq/airbyte/actions/runs/2601310570 (FAILED)
```
You can see that 2 runs were FAILED because of amazon rate-limit.

### STATISTICS FOR LAST WEEK:
2022-07-11T11:07:50Z https://github.com/airbytehq/airbyte/actions/runs/2649117483
2022-07-11T11:07:49Z https://github.com/airbytehq/airbyte/actions/runs/2649117464
2022-07-11T11:07:49Z https://github.com/airbytehq/airbyte/actions/runs/2649117446
2022-07-11T11:07:49Z https://github.com/airbytehq/airbyte/actions/runs/2649117396
2022-07-11T11:07:49Z https://github.com/airbytehq/airbyte/actions/runs/2649117360
2022-07-11T11:07:48Z https://github.com/airbytehq/airbyte/actions/runs/2649117327
2022-07-11T11:07:47Z https://github.com/airbytehq/airbyte/actions/runs/2649117253
2022-07-11T11:07:47Z https://github.com/airbytehq/airbyte/actions/runs/2649117242
2022-07-11T11:07:46Z https://github.com/airbytehq/airbyte/actions/runs/2649117157
2022-07-11T11:07:46Z https://github.com/airbytehq/airbyte/actions/runs/2649117135
2022-07-11T11:07:46Z https://github.com/airbytehq/airbyte/actions/runs/2649117130
2022-07-11T11:07:45Z https://github.com/airbytehq/airbyte/actions/runs/2649117061
2022-07-11T11:07:44Z https://github.com/airbytehq/airbyte/actions/runs/2649116977
2022-07-10T11:07:43Z https://github.com/airbytehq/airbyte/actions/runs/2644402877
2022-07-09T11:07:52Z https://github.com/airbytehq/airbyte/actions/runs/2640871118
2022-07-06T11:08:11Z https://github.com/airbytehq/airbyte/actions/runs/2622345298
2022-07-06T11:08:08Z https://github.com/airbytehq/airbyte/actions/runs/2622345083
2022-07-04T11:08:19Z https://github.com/airbytehq/airbyte/actions/runs/2609646396
2022-07-04T11:08:18Z https://github.com/airbytehq/airbyte/actions/runs/2609646338
2022-07-04T11:08:18Z https://github.com/airbytehq/airbyte/actions/runs/2609646320
2022-07-04T11:08:18Z https://github.com/airbytehq/airbyte/actions/runs/2609646295
2022-07-04T11:08:17Z https://github.com/airbytehq/airbyte/actions/runs/2609646230
2022-07-04T11:08:14Z https://github.com/airbytehq/airbyte/actions/runs/2609645885
2022-07-04T11:08:12Z https://github.com/airbytehq/airbyte/actions/runs/2609645641
2022-07-03T11:09:50Z https://github.com/airbytehq/airbyte/actions/runs/2604850135

## Describe the solution you’d like
I think we need to re-try such failed workflows

### PR DESCRIPTION
**Pull request by grubberr**
Signed-off-by: Sergey Chvalyuk <grubberr@gmail.com>

## What
1. Rewrite shell script `ci_integration_workflow_launcher.sh` to python script `ci_integration_workflow_launcher.py`

2. add UUID for workflow `integration-test` to track progress
    https://stackoverflow.com/questions/69479400/get-run-id-after-triggering-a-github-workflow-dispatch-event
    
3. `ci_integration_workflow_launcher.py` run all workflows, sleep for 20 minutes and re-run AWS failed workflows again

4. `tools/bin/build_report.py` - date added for report

### CODE REVIEW

File: tools/bin/ci_integration_workflow_launcher.py
Comment (grubberr) [reply tools/bin/ci_integration_workflow_launcher.py]: search_workflow_runs -> search_failed_workflow_runs

File: tools/bin/ci_integration_workflow_launcher.py
Comment (grubberr) [reply tools/bin/ci_integration_workflow_launcher.py:44]: yes good catch :)
We assume that step named `Start AWS Runner` will be in specific place of workflow source code.
If workflow source code will be updated and AWS step will move to another place this function will be not actual and we will need to update it.

**Review by lazebnyi — APPROVED** [review]

File: tools/bin/ci_integration_workflow_launcher.py
Comment (edgao) [inline tools/bin/ci_integration_workflow_launcher.py:174]: does this mean that if a connector takes longer than 20 minutes to run, it won't be retried? Are we OK with that?

is it possible to have this logic in the integration-test workflow (i.e. maybe it could accept a `auto_retry_count` option or something, and if it fails then it re-triggers itself with one less retry count)? Then we wouldn't need to have this python script doing a long sleep here either.

**Review by edgao — COMMENTED** [review]
some couple general questions (I'm definitely not a python expert, so not doing a very deep review):

* is `ci_integration_workflow_launcher.sh` still needed, or can we delete it entirely?
* seems like the goal of this PR is to add a single retry to the daily connector tests, is that correct?
* have you tried testing this (i.e. manually trigger the action from this branch)?

**Comment by grubberr** [comment]
> * is `ci_integration_workflow_launcher.sh` still needed, or can we delete it entirely?
yes, I have removed

> * seems like the goal of this PR is to add a single retry to the daily connector tests, is that correct?
yes exactly! 

> * have you tried testing this (i.e. manually trigger the action from this branch)?
yes I have tested changing `master` branch on `grubberr/14450-connector-integration-tests`

File: tools/bin/ci_integration_workflow_launcher.py
Comment (grubberr) [reply tools/bin/ci_integration_workflow_launcher.py:174]: In this script I try to restart ONLY workflows which failed on `AWS Start` step - only this error
If workflows failed on this problem it usually take up to 5 minutes.

**Review by edgao — APPROVED** [review]
this makes sense to me, but let's wait for a review from either topher or pedro under the assumption that they have better-informed opinions about this system than I do

File: tools/bin/ci_integration_workflow_launcher.py
Comment (grubberr) [reply tools/bin/ci_integration_workflow_launcher.py:174]: oops sorry I saw your question only now
yes retries only happens for "AWS Start" problem and only for this problem

File: tools/bin/ci_integration_workflow_launcher.py
Comment (supertopher) [inline tools/bin/ci_integration_workflow_launcher.py:136]: no problem with this timer length, but any logic behind the choice?

**Review by supertopher — APPROVED** [review]
this looks great to my eyes. I appreciated the add.

I assume this is hard to test given that we are catching a transient error.  Please keep an eye on some workflows when merge this to make sure we don't do anything too crazy

sorry for the late review.  feel free to ping me in the future if I'm blocking you :)

**Review by midavadim — APPROVED** [review]

File: .github/workflows/connector_integration_tests.yml
Comment (grubberr) [reply .github/workflows/connector_integration_tests.yml:21]: ok thank you I will check this

File: tools/bin/ci_integration_workflow_launcher.py
Comment (grubberr) [reply tools/bin/ci_integration_workflow_launcher.py:136]: We scan all workflows backwards until we find all needed by uuid
This 3 hours is just additinal protection (no any reason just 3 hours :)

If we did not  found all needed workflows by uuid for some reason
just break loop when we starting to reach "old" (older 3 hours ) workflows

**Pull request merged**
